src/Exception/BatchException.php

```diff
@@ -2,61 +2,49 @@
 
 namespace Http\Client\Exception;
 
+use Http\Client\BatchResult;
+
 /**
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-final class BatchException extends TransferException
+final class BatchException extends RuntimeException
 {
     /**
-     * @var TransferException[]
+     * @var BatchResult
      */
-    private $exceptions;
+    private $result;
 
     /**
-     * @param TransferException[] $exceptions
+     * @param BatchResult $result
      */
-    public function __construct(array $exceptions = [])
+    public function __construct(BatchResult $result)
     {
-        parent::__construct('An error occurred when sending multiple requests.');
-
-        foreach ($exceptions as $e) {
-            if (!$e instanceof TransferException) {
-                throw new InvalidArgumentException('Exception is not an instanceof Http\Client\Exception\TransferException');
-            }
-        }
-
-        $this->exceptions = $exceptions;
+        $this->result = $result;
     }
 
     /**
-     * Returns all exceptions
+     * Returns successful responses or null
      *
-     * @return TransferException[]
+     * @return BatchResult
      */
-    public function getExceptions()
+    public function getResult()
     {
-        return $this->exceptions;
+        return $this->result;
     }
 
     /**
-     * Checks if a specific exception exists
+     * Decides whether the result has any failures
      *
-     * @param TransferException $exception
+     * @param BatchResult $result
      *
-     * @return boolean TRUE if there is the exception else FALSE.
+     * @return BatchResult|BatchException
      */
-    public function hasException(TransferException $exception)
+    public static function decideReturnValue(BatchResult $result)
```

Comment:
not sure if we need this. is the idea to avoid repeating logic in the adapters? while we are at it, we could have a second parameter to decide whether to throw the BatchException or return it. otherwise you would need an instanceof check if you want to throw, which would be weird.

and it would be even more weird to *return* a BatchException from the client. 

oh, but apart from that, i think the decision of throw vs return can be moved to the adapter configuration. so we just say Client::sendRequests either returns a successful BatchResult and you can configure the client whether you want it to throw a BatchException or return a BatchResult. hm. which sounds like a huge pitfal and a pain for generic code that does not know whether the user will configure the client to throw or return... so with this i think we have to make up our mind and decide for either the exception or not.

my proposition is to say the client does the check for $result->hasExceptions() and if there are exceptions instantiates the BatchException and throws it. that way there is no ambiguity. and if i do have the same logic for both situations, in my application i do:

```
try {
    $batchResponse = $client->sendRequests(...);
} catch (BatchException $e) {
    $batchResponse = $e->getBatchResponse();
}
...
```

the alternative is that if i want an exception, the client would have to be written as
```
$batchResponse = $client->sendRequests(...);
if ($batchResponse->hasExceptions()) {
    throw new MyException;
}
...
```

both seem viable and not too much work. but i feel the throwing an exception is more consistent with the sendRequest signature.